### PR TITLE
Bug/1737588/graphics telemetry none not iterable

### PR DESCRIPTION
**Pull request by kik-kik**
I think this should fix the issue. In short I believe during one of the iterations we accidentally set a value of a key to be `None` instead of an actual value or empty dict. Then when we try to check membership we get `'NoneType' is not iterable` exception.

During value assignment now we default to an empty dict if the value we try to assign is `None`.

### CODE REVIEW

**Comment by kik-kik** [comment]
Initially I had a try, except and logged out the values of `k` and `v`, but in this case I don't actually think it would provide any value in debugging as I believe it's one of the previous iterations that actually causes the behaviour we're seeing. We could always log out the actual value of `newdict` at the time exception is being thrown, but also not sure how large that object at this point could be.

**Comment by kik-kik** [comment]
cc: @mikokm Could you help us merging and deploying this if Jeff is happy with the change?

Thanks! 🙇

File: analyses/bigquery_shim/bigquery_shim/dashboard.py [hunk 1]
```diff
@@ -146,11 +146,12 @@ def convert_bigquery_results(f):
...
     newdict = {}
+
     for k, v in d.items():
-        pieces = list(map(revert, k.split("__")))
-        # print(pieces)
+        pieces = list(map(revert, k.split("__")))  # list of values such as clientId, buildId
+
         if len(pieces) == 1:
-            newdict[pieces[0]] = v
+            newdict[pieces[0]] = v if v else {}
```
Comment (jklukas) [inline analyses/bigquery_shim/bigquery_shim/dashboard.py:154]: Another option would be:

```suggestion
            newdict[pieces[0]] = v or {}
```

This syntax is a bit more compact and avoids referring to `v` twice. But it's probably less obvious than the `v if v else {}` construction. I'm leaning towards keeping this the way you've expressed it here.

**Review by jklukas — APPROVED** [review]
Thanks for continuing this investigation! I do expect this will address a variety of possible structural error cases.

File: analyses/bigquery_shim/bigquery_shim/dashboard.py (hunk 1 above)
Comment (kik-kik) [reply analyses/bigquery_shim/bigquery_shim/dashboard.py:154]: oh to be honest I did not realise `v or {}` is valid syntax. Apparently in some languages this is called `elvis operator`([source](https://stackoverflow.com/questions/4978738/is-there-a-python-equivalent-of-the-c-sharp-null-coalescing-operator)) which I did not know.

I kind of like the `or` approach it does feel a bit cleaner...

Comment (kik-kik) [reply analyses/bigquery_shim/bigquery_shim/dashboard.py:154]: ok I guess PR got merged in the meantime, I'll remember or for value setting for the future though. Thanks @jklukas

**Pull request merged**